src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStoragePublisher.java

```diff
@@ -216,12 +223,32 @@ public String getStorageCredentialId() {
     }
 
     public AzureCredentials.StorageAccountCredential getStorageCreds() {
-
-        if (storageCreds == null && storageCredentialId != null) {
-            return AzureCredentials.getStorageAccountCredential(this.storageCredentialId);
+        String credentialsId = this.storageCredentialId == null
+                ? (this.storageCreds == null ? null : this.storageCreds.getId())
+                : this.storageCredentialId;
+        try {
+            if (credentialsId != null) {
+                AzureCredentials.StorageAccountCredential credentials = AzureCredentials.getStorageAccountCredential(credentialsId);
+                if (credentials != null) {
+                    this.storageCreds = credentials;
+                    return credentials;
+                }
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
         }
-
-        return storageCreds;
+        if (this.storageCreds == null) {
```

Comment:
I think I can simplify this ..  Initially I started writing this piece of code while following the migration examples of other plugins, but now I'm migrating differently, so I can modify accordingly